dspace-api/src/main/java/org/dspace/util/SolrImportExport.java

```diff
@@ -0,0 +1,633 @@
+				                                              " By default, the contents of this directory will be deleted once the reindex has finished." +
+				                                              " Ignored when action is export or import.");
+		options.addOption(LAST_OPTION, "last", true, "When exporting, export records from the last [timeperiod] only." +
+				                                             " This can be one of: 'd' (beginning of yesterday through to now);" +
+				                                             " 'm' (beginning of the previous month through to end of the previous month);" +
+				                                             " a number, in which case the last [number] of days are exported, through to now (use 0 for today's data)." +
+															 " Date calculation is done in UTC. If omitted, all documents are exported.");
+		return options;
+	}
+
+	/**
+	 * Reindexes the specified core
+	 *
+	 * @param indexName the name of the core to reindex
+	 * @param exportDir the directory to use for export
+	 * @param keepExport whether to keep the contents of the exportDir after the reindex. If false, this directory will be deleted at the end of the reimport.
+	 */
+	private static void reindex(String indexName, String exportDir, boolean keepExport) throws IOException, SolrServerException, SolrImportExportException {
+		String tempIndexName = indexName + "-temp";
+
+		String origSolrUrl = makeSolrUrl(indexName);
+		String baseSolrUrl = StringUtils.substringBeforeLast(origSolrUrl, "/"); // need to get non-core solr URL
+		String tempSolrUrl = baseSolrUrl + "/" + tempIndexName;
+
+		String solrInstanceDir = ConfigurationManager.getProperty("dspace.dir") + File.separator + "solr" + File.separator + indexName;
+		// the [dspace]/solr/[indexName]/conf directory needs to be available on the local machine for this to work
+		// -- we need access to the schema.xml and solrconfig.xml file, plus files referenced from there
+		// if this directory can't be found, output an error message and skip this index
+		File solrInstance = new File(solrInstanceDir);
+		if (!solrInstance.exists() || !solrInstance.canRead() || !solrInstance.isDirectory())
+		{
+			throw new SolrImportExportException("Directory " + solrInstanceDir + "/conf/ doesn't exist or isn't readable." +
+					                   " The reindexing process requires the solr configuration directory for this index to be present on the local machine" +
+					                   " even if the Solr is running on a different host. Not reindexing index " + indexName);
+		}
+
+		String timeField = makeTimeField(indexName);
+
+		// Create a temp directory to store temporary core data
+		File tempDataDir = new File(ConfigurationManager.getProperty("dspace.dir") + File.separator + "temp" + File.separator + "solr-data");
+		boolean tempDirCreated = tempDataDir.mkdirs();
+		if (!tempDirCreated)
```

Comment:
Note:
Running ```./dspace solr-reindex-statistics``` as e.g., root, creates temp/ directory with too strict permissions and tomcat will not be able to write into it.
If you create the temp/solr-data directory beforehand, it will throw exception at this line.